BRANCHES.md

```diff
@@ -28,6 +28,15 @@ undocumented features, then you should be able to re-compile it without
 modification with any later release x.y'.z' with the same major version
```

Comment:
Pre-existing: maybe it would be helpful to separate API and the ABI compatibility discussions to their respective subsections.